### PR TITLE
Make function std::ping::ping concurrency safe

### PR DESCRIPTION
**Pull request by michael-hartmann**
The function `ping` so far did not check the ident received. As a consequence the function `ping` might returned wrong results when used simultaneously from different threads, see https://github.com/aisk/ping/issues/6 for more details.

This commit fixes the issue by checking the ident of the reply and comparing it with the expected ident.

Changes:
* Changed type of `timeout` from `Option<Duration>` to `Duration`. This avoids an additional error check in new code.
* Receive echo in a loop until either the correct ident was received or a timeout occured.

I have tested the changes and they seem to work for me. Let me know if something still needs to be changed.

### CODE REVIEW

File: src/ping.rs [hunk 1]
```diff
@@ -43,30 +45,44 @@ pub fn ping(addr: IpAddr, timeout: Option<Duration>, ttl: Option<u32>, ident: Op
...
+
+        if reply.ident == request.ident {
+            // received correct ident
+            return Ok(());
         }
-    };
 
-    return Ok(());
+        // if ident is not correct check if timeout is over
+        time_elapsed = SystemTime::now().duration_since(time_start).expect("Clock may have gone backwards");
```
Comment (aisk) [inline src/ping.rs:82]: Since this is a library, call `expect` is not good here, for it may cause panic for user's process. Return an error will be good.

**Comment by aisk** [comment]
Great work! I have only a small issue which is pointed out up here. And it should be better to add a mutiple thread test case like what you posted on #6

**Comment by michael-hartmann** [comment]
Thanks for the review, sounds fine. I am currently a bit occupied, give me a couple of days to update the PR.

**Comment by aisk** [comment]
These commits is already merged in #9.

**Pull request closed without merging**